### PR TITLE
feat: refactor saves/list inclusion properties to defer until needed; replace with `isSavedToAnyList`

### PR DESCRIPTION
**Pull request by mzikherman**
### The setup

Right now, we typically request two properties - `isSaved` and `isSavedToList` whenever we are fetching artwork 'save' information. These lead to two similar simultaneous Postgres queries on the backend - the top 2 most common Postgres queries Gravity sees. We fetch this for every artwork in a grid/rail.

What do these fields mean?

`isSaved` - is the artwork saved to the default 'saves' collection
`isSavedToList` - is the artwork saved to any non-default (custom) collection that is considered a 'saves' collection (so _not_ disliked artworks)

### What the PR does

This PR updates fragments where both of these properties are being requested, in favor of a new _single_ property - `isSavedToAnyList` (added in https://github.com/artsy/metaphysics/pull/6190).

This winds up just making one Postgres query, and so while this is all unlikely to yield any real user-facing performance improvement as the _two_ queries were parallelized (and fast) - it does lessen the load on Postgres.

### What's the catch?

The obvious question is - why we were querying both of these fields up front - what behavior was occurring in the UI that depended on these fields? Is it possible to replace them like this without breaking that functionality?

With one tiny exception (which I'll note inline) - the answer is: **the only behavior for a signed-in user that depended on those two properties was when un-saving a work that was _also_ saved to a custom list - that popped up a list management modal rather than simply unsaving**.

Thus, the main code update here to support replacing `isSaved` and `isSavedToList` with the singular `isSavedToAnyList` is that, on an un-save, we need to fetch the list inclusion information to know whether or not to pop-up the modal. Similarly, when a logged-out user clicks a save button, after login we fetch the comprehensive list inclusion information to know how to direct the newly logged-in user.

### CODE REVIEW

File: src/Components/Artwork/SaveButton/useSaveArtworkToLists.ts [hunk 1]
```diff
@@ -115,21 +114,41 @@ export const useSaveArtworkToLists = (options: SaveArtworkToListsOptions) => {
...
     }
 
-    if (savedListId || artwork.isSavedToCustomLists) {
-      openSelectListsForArtworkModal()
-      return ResultAction.SelectingListsForArtwork
-    }
-
-    const response = await saveToDefaultList()
-    const isSaved = response?.saveArtwork?.artwork?.isSaved
+    if (!artwork.isSavedToAnyList) {
```
Comment (mzikherman) [inline src/Components/Artwork/SaveButton/useSaveArtworkToLists.ts:117]: This is the 'meat' of the PR where the 'unsave' flow has shifted to first fetch custom list inclusion status to know whether or not to pop up the modal.

File: src/Components/Artwork/useArtworkLists.ts [hunk 2]
```diff
@@ -28,18 +30,37 @@ export const useArtworkLists = (options: SaveArtworkToListsOptions) => {
...
+              isSavedToList
+              isSaved
+            }
+          }
+        `,
+        { artworkID: artwork.internalID }
+      ).toPromise()
+
+      // The user clicked the save button while logged out.
+      // We need to comprehensively check the artwork's save state.
```
Comment (mzikherman) [inline src/Components/Artwork/useArtworkLists.ts:48]: This is analogous to the code change above (checking list inclusion on 'unsave'). This branch is for when someone clicks an (unsaved) save button while logged out - then they login - what action to take after login.

The artwork in question might be saved by them already to a custom list, _just_ to their default saves, or not saved at all.

So similarly, we want to comprehensively fetch the current saved state to direct them properly.

File: src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx [hunk 3]
```diff
@@ -78,17 +77,17 @@ export const ArtworkActionsSaveButton: FC<ArtworkActionsSaveButtonProps> = ({
   if (isOpenOrUpcomingSale) {
     return (
       <ArtworkActionsWatchLotButtonFragmentContainer
-        isSaved={!!artwork.isSaved}
+        isSaved={artwork.isSavedToAnyList}
         artwork={artwork}
         onClick={handleSaveArtworkInAuction}
-        canShowRegistrationPopover={!artwork.isSavedToList}
+        canShowRegistrationPopover
```
Comment (mzikherman) [inline src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx:83]: This is the one exception mentioned in the PR description - it wasn't clear to me why we might suppress a bidder registration popover when an artwork _is_ saved to a custom list (potentially in _addition_ to being saved in the default collection). So just rolled with `true`.

The popover itself only shows up if you started out clicking the save button while logged-out: https://github.com/artsy/force/blob/061f45bcb93fd9fee41d1ec2594798928cb397fa/src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsWatchLotButton.tsx#L65

**Review by damassi — DISMISSED** [review]
Code changes are straightforward 👍

**Review by joeyAghion — DISMISSED** [review]

File: src/Components/Artwork/SaveButton/useSaveArtworkToLists.ts [hunk 4]
```diff
@@ -115,21 +114,41 @@ export const useSaveArtworkToLists = (options: SaveArtworkToListsOptions) => {
...
+          query useSaveArtworkToListsArtworkListInclusionQuery(
+            $artworkID: String!
+          ) {
+            artwork(id: $artworkID) {
+              isSavedToList(default: false)
+            }
+          }
+        `,
+        { artworkID: artwork.internalID }
+      ).toPromise()
```
Comment (anandaroop) [inline src/Components/Artwork/SaveButton/useSaveArtworkToLists.ts:139]: This `fetchQuery(…).toPromise()` dance is kind of a new one for me — so this is how we imperatively issue Relay queries outside of the normal component lifecycle, yeah? GTK

**Review by anandaroop — DISMISSED** [review]
👍🏽

File: src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx (hunk 3 above)
Comment (mzikherman) [reply src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveButton.tsx:83]: Yes - by 'save' i mean 'watch lot' in the context of this auction thing - whoops!

**Pull request merged**